sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExec.scala

```diff
@@ -153,11 +153,15 @@ sealed abstract class GpuMetric extends Serializable {
   def add(v: Long): Unit
 
   final def ns[T](f: => T): T = {
+    val needTrack = ThreadLocalMetrics.onMetricsEnter(this)
```

Comment:
Seems `start` is not needed if `needTrack` is `false`.  Or do we really not compute the metric when `needTrack` is false ?
```
val start = if (needTrack) {
  Some(System.nanoTime())
} else None


...
  } finally {
    start.foreach { startTime =>
      add(System.nanoTime() - startTime)
      ThreadLocalMetrics.onMetricsExit(this)
  }     
```

Reply:
yes, since the cost of System.nanoTime() is neglectable, I kept the original structure